src/edu/washington/escience/myria/operator/SymmetricHashJoin.java

```diff
@@ -331,7 +331,13 @@ public SymmetricHashJoin(final List<String> outputColumns, final Operator left,
   @Override
   protected Schema generateSchema() {
     final Schema leftSchema = getLeft().getSchema();
+    if (leftSchema == null) {
```

Comment:
The situation is, it's possible to get `null` here if this join depends on, for example, a consumer that depends on a producer which doesn't have its schema to be set yet. Then we will have `NullPointerException` in this line following:
`Type leftType = leftSchema.getColumnType(leftIndex);`